### ISSUE TITLE
Encoding problems on Windows computers

### ISSUE THREAD

**Issue by Quadric** [thinkwelltwd/device_detector#30]
On version 0.9 worked ok, but on 0.10 i have a following exception when i try to parse user_agent string (the "x" variable):

```
In [1]: x = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, 
   ...:  like Gecko) Chrome/83.0.4103.61 Safari/537.36'

In [2]: from device_detector import DeviceDetector

In [3]: result1 = DeviceDetector(x).parse()
---------------------------------------------------------------------------
UnicodeDecodeError                        Traceback (most recent call last)       
<ipython-input-3-55b0fb999d44> in <module>
----> 1 result1 = DeviceDetector(x).parse()

~\AppData\Local\Continuum\miniconda3\envs\bluebooster_sandbox\lib\site-packages\device_detector\device_detector.py in parse(self)
    231 
    232         if not self.skip_bot_detection:
--> 233             self.parse_bot()
    234             if self.is_bot():
    235                 return self.set_parse_cache()

~\AppData\Local\Continuum\miniconda3\envs\bluebooster_sandbox\lib\site-packages\device_detector\device_detector.py in parse_bot(self)
    310         """
    311         if not self.skip_bot_detection and not self.bot:
--> 312             self.bot = Bot(self.user_agent, self.ua_hash, self.ua_spaceless).parse()
    313             self.all_details['bot'] = self.bot.ua_data
    314

~\AppData\Local\Continuum\miniconda3\envs\bluebooster_sandbox\lib\site-packages\device_detector\parser\parser.py in parse(self)
     85             return self
     86
---> 87         self._parse()
     88         self.extract_details()
     89 

~\AppData\Local\Continuum\miniconda3\envs\bluebooster_sandbox\lib\site-packages\device_detector\parser\parser.py in _parse(self)
     69     def _parse(self) -> None:
     70         """Override on subclasses if custom parsing is required"""
---> 71         for ua_data in self.regex_list:
     72             match = self._check_regex(ua_data['regex'])
     73             if match:

~\AppData\Local\Continuum\miniconda3\envs\bluebooster_sandbox\lib\site-packages\device_detector\yaml_loader.py in regex_list(self)
     79
     80         for fixture in self.fixture_files:
---> 81             regexes.extend(self.yaml_to_list('regexes/{}'.format(fixture)))
     82
     83         for regex in regexes:

~\AppData\Local\Continuum\miniconda3\envs\bluebooster_sandbox\lib\site-packages\device_detector\yaml_loader.py in yaml_to_list(self, yfile)
     61         Load yaml file to list of dicts
     62         """
---> 63         regexes = self.load_from_yaml(yfile)
     64         if isinstance(regexes, list):
     65             return regexes

~\AppData\Local\Continuum\miniconda3\envs\bluebooster_sandbox\lib\site-packages\device_detector\yaml_loader.py in load_from_yaml(yfile)
     33         if Path('{}/{}'.format(ROOT, yfile)).exists():
     34             with open('{}/{}'.format(ROOT, yfile), 'r') as yf:
---> 35                 return yaml.load(yf, SafeLoader)
     36
     37         try:

~\AppData\Local\Continuum\miniconda3\envs\bluebooster_sandbox\lib\site-packages\yaml\__init__.py in load(stream, Loader)
    112     loader = Loader(stream)
    113     try:
--> 114         return loader.get_single_data()
    115     finally:
    116         loader.dispose()

~\AppData\Local\Continuum\miniconda3\envs\bluebooster_sandbox\lib\site-packages\yaml\constructor.py in get_single_data(self)
     47     def get_single_data(self):
     48         # Ensure that the stream contains a single document and construct 
it.
---> 49         node = self.get_single_node()
     50         if node is not None:
     51             return self.construct_document(node)

ext\_yaml.pyx in _yaml.CParser.get_single_node()

ext\_yaml.pyx in _yaml.CParser._compose_document()

ext\_yaml.pyx in _yaml.CParser._compose_node()

ext\_yaml.pyx in _yaml.CParser._compose_sequence_node()

ext\_yaml.pyx in _yaml.CParser._compose_node()

ext\_yaml.pyx in _yaml.CParser._compose_mapping_node()

ext\_yaml.pyx in _yaml.CParser._compose_node()

ext\_yaml.pyx in _yaml.CParser._parse_next_event()

ext\_yaml.pyx in _yaml.input_handler()

~\AppData\Local\Continuum\miniconda3\envs\bluebooster_sandbox\lib\encodings\cp1250.py in decode(self, input, final)
     21 class IncrementalDecoder(codecs.IncrementalDecoder):
     22     def decode(self, input, final=False):
---> 23         return codecs.charmap_decode(input,self.errors,decoding_table)[0] 
     24
     25 class StreamWriter(Codec,codecs.StreamWriter):

UnicodeDecodeError: 'charmap' codec can't decode byte 0x83 in position 4159: character maps to <undefined>
```

Python 3.7.7
Windows 10
device_detector 0.10.0

on version 0.9 there is no problem.

Looks like a problem with encoding?

**Comment by thinkwelltwd** [comment]
I'm not a Windows developer so I can't answer your question.  Version .10 is working for me on Linux and macOS.

**Comment by elias-shipt** [comment]
I had the same issue on my windows, using jupyter.

**Comment by thinkwelltwd** [comment]
I'm sure you did. I don't work on Windows, so I'm not in a position to fix it. PRs welcome.

**Comment by thinkwelltwd** [comment]
Fixed by #32
